### DISCUSSION TITLE
Purpose of training dataset in testing?

### DISCUSSION THREAD

**Discussion by ozanbarism** [SalesforceAIResearch/uni2ts#79] [General]
Hi, i was checking your starter code shown below. 

# Split into train/test set
train, test_template = split(
    ds, offset=-TEST
)  # assign last TEST time steps as test set

# Construct rolling window evaluation
test_data = test_template.generate_instances(
    prediction_length=PDT,  # number of time steps for each prediction
    windows=TEST // PDT,  # number of windows in rolling window evaluation
    distance=PDT,  # number of time steps between each window - distance=PDT for non-overlapping windows
)

# Prepare pre-trained model by downloading model weights from huggingface hub
model = MoiraiForecast(
    module=MoiraiModule.from_pretrained(f"Salesforce/moirai-1.0-R-{SIZE}"),
    prediction_length=PDT,
    context_length=CTX,
    patch_size=PSZ,
    num_samples=100,
    target_dim=1,
    feat_dynamic_real_dim=ds.num_feat_dynamic_real,
    past_feat_dynamic_real_dim=ds.num_past_feat_dynamic_real,
)

predictor = model.create_predictor(batch_size=BSZ)
forecasts = predictor.predict(test_data.input)

here the train is never used again. Additionally, the predictor takes test_data as input. however, i believe it should take some context window from training data as input and then make predictions that will be compared to the test data. how can i achieve that? what do i need to change?

**Comment by gorold** [comment]
test_data.input contains the context windows, whereas test_data.labels contain the forecast horizon, no need to make any changes.
